compiler+runtime/src/cpp/jank/runtime/obj/persistent_string_sequence.cpp

```diff
@@ -0,0 +1,113 @@
+#include <jank/runtime/obj/persistent_string_sequence.hpp>
+
+namespace jank::runtime
+{
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr s)
+    : str{ s }
+  {
+    assert(s->count() > 0);
+  }
+
+  obj::persistent_string_sequence::static_object(obj::persistent_string_ptr s, size_t i)
+    : str{ s }
+    , index{ i }
+  {
+    assert(s->count() > 0);
+  }
+
+  /* behavior::objectable */
+  native_bool obj::persistent_string_sequence::equal(object const &o) const
+  {
+    if(o.type != object_type::persistent_string_sequence)
+    {
+      return false;
+    }
+    auto const seq(expect_object<obj::persistent_string_sequence>(&o));
+    return std::equal(str->data.begin() + index,
+                      str->data.end(),
+                      seq->str->data.begin() + seq->index,
+                      seq->str->data.end());
+  }
+
+  void obj::persistent_string_sequence::to_string(fmt::memory_buffer &buff) const
+  {
+    return behavior::detail::to_string(
+      str->data.begin()
+        + static_cast<decltype(obj::persistent_string::data)::difference_type>(index),
```

Comment:
Why was this cast needed, rather than just doing `+ index` (like you did on line 26)?